src/AspNet.Security.OAuth.Introspection/OAuthIntrospectionHandler.cs

```diff
@@ -146,48 +203,29 @@ public class OAuthIntrospectionHandler : AuthenticationHandler<OAuthIntrospectio
             return JObject.Parse(await response.Content.ReadAsStringAsync());
         }
 
-        protected virtual Task<bool> ValidateAudienceAsync(JObject payload) {
+        protected virtual Task<bool> ValidateAudienceAsync(AuthenticationTicket ticket)
+        {
             // If no explicit audience has been configured,
             // skip the default audience validation.
-            if (Options.Audiences.Count == 0) {
+            if (Options.Audiences.Count == 0)
+            {
                 return Task.FromResult(true);
             }
 
-            // If no "aud" claim was returned by the authorization server,
-            // assume the access token was not specific enough and reject it.
-            if (payload[OAuthIntrospectionConstants.Claims.Audience] == null) {
+            // Extract the audiences from the authentication ticket.
+            string audiences;
+            if (!ticket.Properties.Items.TryGetValue(OAuthIntrospectionConstants.Properties.Audiences, out audiences))
```

Comment:
:+1: I like it.

Just don't forget to add the logic extracting the `aud` claim (string/array) and storing the audiences as authentication property or all the related tests will keep failing hard :smile: